src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import userEvent from '@testing-library/user-event'
+
+import { SpouseInfo } from './SpouseAndDependent'
+import { store } from '../../redux/store'
+
+afterEach(async () => {
+  await waitFor(() => localStorage.clear())
+  jest.resetAllMocks()
+})
+
+describe('SpouseInfo', () => {
+  it('renders an `Add` button when no spouse has been added', () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
```

Comment:
I'd like to put together a helper render function that takes in a Provider to abstract some of this away.